File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/ImageFragmentItemFactory.java
package me.xiaopan.sketchsample.adapter.itemfactory;

import android.content.Context;
import android.support.v4.app.Fragment;

import me.xiaopan.assemblyadapter.AssemblyFragmentItemFactory;
import me.xiaopan.sketchsample.bean.Image;
import me.xiaopan.sketchsample.fragment.ImageFragment;
import me.xiaopan.sketchsample.util.AppConfig;

public class ImageFragmentItemFactory extends AssemblyFragmentItemFactory<Image> {
    private Context context;
    private String loadingImageOptionsId;

    public ImageFragmentItemFactory(Context context, String loadingImageOptionsId) {
        this.context = context;
        this.loadingImageOptionsId = loadingImageOptionsId;
    }

    @Override
    public boolean isTarget(Object o) {
        return o instanceof Image;
    }

    @Override
    public Fragment createFragment(int i, Image image) {
        boolean showTools = AppConfig.getBoolean(context, AppConfig.Key.SHOW_TOOLS_IN_IMAGE_DETAIL);
        return ImageFragment.build(image, loadingImageOptionsId, showTools);
    }
}

File: sample/src/main/java/me/xiaopan/sketchsample/adapter/itemfactory/ImageFragmentItemFactory.kt
package me.xiaopan.sketchsample.adapter.itemfactory

import android.content.Context
import android.support.v4.app.Fragment

import me.xiaopan.assemblyadapter.AssemblyFragmentItemFactory
import me.xiaopan.sketchsample.bean.Image
import me.xiaopan.sketchsample.fragment.ImageFragment
import me.xiaopan.sketchsample.util.AppConfig

class ImageFragmentItemFactory(private val context: Context, private var loadingImageOptionsId: String?) : AssemblyFragmentItemFactory<Image>() {

    override fun isTarget(o: Any): Boolean {
        return o is Image
    }

    override fun createFragment(i: Int, image: Image): Fragment {
        val showTools = AppConfig.getBoolean(context, AppConfig.Key.SHOW_TOOLS_IN_IMAGE_DETAIL)
        return ImageFragment.build(image, loadingImageOptionsId, showTools)
    }
}
